core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -649,6 +652,10 @@ boolean isPreserveAliases() {
                     wildcardSeen = true;
                 }
             }
+            if(isDeprecated) {
+              new DeprecationLogger(Loggers.getLogger(IndexNameExpressionResolver.class))
```

Comment:
any reason why the deprecation logger is created on demand each time rather than a static field?